### ISSUE TITLE
Design for supporting rpm-ostree jigdo ♲📦

### ISSUE THREAD

**Issue by cgwalters** [rhinstaller/anaconda#1259]
See this issue: https://github.com/projectatomic/rpm-ostree/issues/1081

This issue will track proposed Anaconda changes to enable this.  Here are my thoughts so far:

kickstart looks like this:

```
ostreesetup --jigdo updates:fedora-atomic-host
```

(The idea is that jigdo is [repo bound](https://github.com/projectatomic/rpm-ostree/pull/1130) by default, hence the `updates:` prefix)

We should also enable defaults for things like `--osname` - part of the whole idea here is to require less learning about ostree.   I think we can do the same as the LVM volume group logic?  The idea is quite similar; it's a system-level thing that needs *a* name.  Or maybe we just say `default`?

One thing I briefly debated is supporting specifying jigdo via the existing `%packages` - which would naturally lead into supporting rpm-ostree package layering, so like:

```
%packages
fedora-atomic-host
libvirt
```

but...that'd get messy as we'd need some way to detect that `fedora-atomic-host` is an rpm-ostree jigdo RPM.  And since we already have the `ostreesetup` verb, I'd vote we reuse that.

**Comment by cgwalters** [comment]
Ah, and I forgot to say, a tricky aspect here is handling the "embedded content" (DVD) case.  I think we can now change the rpmostree-based builds to inject in the same place as traditional (`/run/media/repo` or something)?  I need to look at that.  But it's not strictly required, though doing so would make it work more easily to have a single DVD-like ISO that supports *both* traditional and rpmostree installs.  But that in turn would lead into some UI for choosing, which so far I haven't considered doing since anything touching the Anaconda UI requires lots more design/work.

**Comment by cgwalters** [comment]
One other major thing I forgot here; this is going to finally cause us to deal with the fact that our `rpmostreepayload.py` doesn't really use rpm-ostree much, just ostree.  So an interesting question here is: do we care about people using Anaconda to do installs using "pure ostree", i.e. no RPM involved?

(An analogous situation here is things like the `tarpayload` - that doesn't care about RPM any more that libostree does)

Personally my vote is that we simply hard depend on rpm-ostree, and anyone who cares about the pure libostree case gets to help out with patches.

One major technical consequence of this will be that rather than calling libostree from a thread, we'll mostly be speaking DBus to `rpm-ostree start-daemon --sysroot /mnt/sysroot`.

**Comment by jkonecny12** [comment]
I don't think this should be a problem. Event he file name is `rpmostreepayload.py` not `ostreepayload.py`. In case we need just ostree we can always create new payload with inheritance involved.

**Comment by cgwalters** [comment]
Here I briefly note it's likely possible to use Anaconda to install Debian (or other content) inside an ostree commit: https://mail.gnome.org/archives/ostree-list/2018-April/msg00021.html  Though where things get a lot more uncertain is as soon as one starts using kickstart verbs.

**Comment by poncovka** [comment]
Thank you for reporting this issue! The issue tracker will be closed on 3 June 2019. Please make sure that your issue is also reported at https://bugzilla.redhat.com/. We are handling a high number of issues and many of them involve other components. Bugzilla allows us to better process your issues and collaborate on them with other teams. We are sorry for the inconvenience.
